cosmwasm/packages/wasmi-runtime/src/wasm/errors.rs

```diff
@@ -1,45 +1,27 @@
-#![allow(unused)]
-
 use derive_more::Display;
-use enclave_ffi_types::EnclaveError;
+use enclave_ffi_types::{EnclaveError, UntrustedVmError};
 use wasmi::{Error as InterpreterError, HostError};
 
 #[derive(Debug, Display)]
 #[non_exhaustive]
 pub enum WasmEngineError {
-    FailedOcall,
+    #[display(fmt = "FailedOcall")]
+    FailedOcall(UntrustedVmError),
     OutOfGas,
+    Panic,
+
     EncryptionError,
     DecryptionError,
-    DbError(DbError),
+
     MemoryAllocationError,
     MemoryReadError,
     MemoryWriteError,
```

Comment:
These three variants are really nice and high-level, but they eventually just get ignored. (converted to `EnclaveError::Unknown`). I'd like to instead use variants that are analogous to the variants of `CommunicationError` (see under the `cosmwasm-vm` errors), add similar variants to `EnclaveError`, and then add the two conversions. This way we'll have better integration with the error variants that are already available, and give better diagnostics for misbehaving contracts. What do you think? this is about 20 minutes of work, that i'd gladly do, but i want feedback on it.

Reply:
Lets open a new issue and do it some other time - we need to get this merged already

Reply:
https://github.com/enigmampc/SecretNetwork/issues/331